### PR TITLE
Playlist and tracks

### LINKED ISSUE #39: Playlist feature
### Usecase

As a user I should be able to browse through all my playlists.

### Backend
- Create routes to fetch all your playlists. This route should accept an optional query string in the event the user searches for a playlist.
- Should make an api call to spotify. 
- Inspect the response object and then create an interface on the frontend.

### Frontend
- Create a tab called `Playlist` in the Footer that will have a searchbar and a list of playlists that will be fetched from the above api.
- The playlist should be expandable to see the songs in the playlist.
- On hover a `Select` button should appear that should select the playlist.
- Create interfaces based on the response

**NOTE:** 
 - Should we also implement adding/editing and deleting of playlists from vibey?  @s-munro

### PR DESCRIPTION
**Pull request by hrishikeshpaul**
Issue #39 

This PR has 2 routes,
1. Get all playlists of the user
2. Get all tracks of the user

These 2 routes are going to be helpful when the creator of the room need to browser through playlists/songs. I'm going to start developing the frontend for this shortly.

#### How to test?
1. Use postman and add 2 headers,
   ```
   vb-spotify-at: <spotify access token here>
   vb-spotify-rt: <spotify refresh token here>
   ```

2. To test the tracks, first hit the playlist route, get the id and then use the id to test the tracks route.

[OR]
1. Remove the `setSpotifyTokens` middleware.
2. Login from the frontend
3. Hit the routes from the browser

### CODE REVIEW

File: vb-be/src/routes/playlist.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+const { spotifyApi } = require("../lib/spotify");
+const express = require("express");
```
Comment (s-munro) [inline vb-be/src/routes/playlist.js:4]: just a nit, but let's organize the imports by external modules, then imports.  I'll see if I can set up some import paths for us soon so we can do like we do on the FE

Comment (hrishikeshpaul) [reply vb-be/src/routes/playlist.js:4]: Oh the eslint isn't set up for the backend. The backend is pretty shaky rn lol

Comment (hrishikeshpaul) [reply vb-be/src/routes/playlist.js:4]: I mean eslint is there but it does very less

File: vb-be/src/routes/playlist.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ */
+router.get("/", setSpotifyTokens, async (req, res, next) => {
+  const { limit, offset } = req.query;
+
+  if (!limit || !offset) {
+    res
+      .status(HttpStatus.Error)
+      .send(
+        new ErrorHandler(
+          HttpStatus.Error,
```
Comment (s-munro) [inline vb-be/src/routes/playlist.js:23]: What code does this throw?  Just curious, I've only ever hard-coded errors in

File: vb-be/src/routes/playlist.js (hunk 1 above)
Comment (s-munro) [reply vb-be/src/routes/playlist.js:4]: What do you think we should add to the linter for the BE?  We can arrange that soon though, but we should make sure our imports etc are still being implemented right even without the linter

File: vb-be/src/routes/playlist.js (hunk 2 above)
Comment (s-munro) [reply vb-be/src/routes/playlist.js:23]: Nvm just saw below that you implemented the class lol

**Review by s-munro — CHANGES_REQUESTED** [review]
Dope, this is looking super solid to me, aside from just the imports.  Good to merge after, in my books 👍

**Pull request merged**